src/merge_maps_tool.cpp

```diff
@@ -203,26 +198,26 @@ bool MergeMapTool::MergeMapCallback(slam_toolbox::MergeMaps::Request &req,
       {
         pScan= *iter;
         pScan_copy = pScan;
-        tf::Transform submap_correction = submaps_correction_tf[id];
+        tf::Transform submap_correction = submap_marker_transform_[id];
 
         //TRANSFORM BARYCENTERR POSE
         const karto::Pose2 baryCenter_pose = pScan_copy->GetBarycenterPose();
-        auto karto_baryCenterPose_corr = ApplyCorrection(baryCenter_pose, submap_correction);
+        karto::Pose2 karto_baryCenterPose_corr = ApplyCorrection(baryCenter_pose, submap_correction);
```

Comment:
const ref?

Reply:
actually i really do want to return an object not a reference :D